modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -354,7 +354,7 @@ private void recordFailure(Failure failure, List<Failure> failures) {
      * Start terminating a request that finished non-catastrophically.
      */
     void startNormalTermination(List<Failure> indexingFailures, List<ShardSearchFailure> searchFailures, boolean timedOut) {
-        if (task.isCancelled() || false == mainRequest.isRefresh()) {
+        if (task.isCancelled() || false == mainRequest.isRefresh() || destinationIndices.isEmpty()) {
```

Comment:
Just curious, but when would `destinationIndices` be empty? Wouldn't you always re-index into an index?